transcom/mymove
# MB-6825 update pricer for domestic unpacking (DUPK)

## Description

This PR adds display params for the DUPK service item. It also adds two formatting functions to display bool and float values.

## Setup

`make server_test`

To view the pricer params, follow the steps below:

- Build and start your local client/server (and make sure your DB is up-to-date)
- Create a new move in the customer UI
- As a TOO in the office UI, edit the orders to add the missing fields, then approve the move and the service items. That should automatically add other service items like DUPK.
- Run `prime-api-demo <move code> local` to populate actual weight and other params needed to create a payment request
- Create a payment request using postman with the following body
```
{
    "moveTaskOrderID": "<the move id you created in the UI>",
    "serviceItems": [
        {
            "id": "<ID of the DUPK service item you want to create a payment request for. >"
        }
    ],
    "isFinal": false,
    "pointOfContact": "Duis est id dolore elit"
}
```
- Running a fetch-mto-updates via the prime-api-client or postman. You will see the params in the response:
```
                            {
                                "eTag": "MjAyMS0wNC0wN1QxNjoyMDozMS40MzM2OTNa",
                                "id": "4311f859-d2b4-4bd7-9bd7-e2e4c2e6290b",
                                "key": "ContractYearName",
                                "origin": "PRICER",
                                "paymentServiceItemID": "a26df49d-f817-4c28-93b8-94521516ba3c",
                                "type": "STRING",
                                "value": "Base Period Year 2"
                            },
                            {
                                "eTag": "MjAyMS0wNC0wN1QxNjoyMDozMS40MzgxNzla",
                                "id": "88635d30-237f-4d10-bcec-72b9af8a066a",
                                "key": "EscalationCompounded",
                                "origin": "PRICER",
                                "paymentServiceItemID": "a26df49d-f817-4c28-93b8-94521516ba3c",
                                "type": "DECIMAL",
                                "value": "1.02060"
                            },
                            {
                                "eTag": "MjAyMS0wNC0wN1QxNjoyMDozMS40NDM1MVo=",
                                "id": "e6283e3e-155b-466c-b157-3fa2a0b6a2bc",
                                "key": "IsPeak",
                                "origin": "PRICER",
                                "paymentServiceItemID": "a26df49d-f817-4c28-93b8-94521516ba3c",
                                "type": "BOOLEAN",
                                "value": "false"
                            },
                            {
                                "eTag": "MjAyMS0wNC0wN1QxNjoyMDozMS40NDg1NTNa",
                                "id": "3707f149-4884-4a31-aa23-067097da22c6",
                                "key": "PriceRateOrFactor",
                                "origin": "PRICER",
                                "paymentServiceItemID": "a26df49d-f817-4c28-93b8-94521516ba3c",
                                "type": "DECIMAL",
                                "value": "2.379"
                            }
```
<img width="1085" alt="Screen Shot 2021-04-07 at 2 58 57 PM" src="https://user-images.githubusercontent.com/56279459/113945074-10659e00-97b2-11eb-8b60-746cd56ca1bb.png">

## Code Review Verification Steps

* Any new migrations/schema changes:

## References

* [Jira story](tbd) for this change
* [this article](tbd) explains more about the approach used.

## Screenshots

If this PR makes visible UI changes, an image of the finished UI can help reviewers and casual
observers understand the context of the changes. A before image is optional and
can be included at the submitter's discretion.

Consider using an animated image to show an entire workflow instead of using multiple images. You may want to use GIPHY CAPTURE for this! 📸

_Please frame screenshots to show enough useful context but also highlight the affected regions._
